app/grandchallenge/archives/views.py

```diff
@@ -334,6 +358,153 @@ def get_context_data(self, **kwargs):
+                continue
+            ci = ComponentInterface.objects.get(slug=slug)
+            civ = self.archive_item.values.filter(interface=ci).first()
+            if civ and civ.value is not None:
+                if civ.value == value:
+                    continue
+                civ_pks.remove(civ.pk)
+                civ = None
+            if not civ:
+                civ = ComponentInterfaceValue.objects.create(interface=ci)
+            if ci.kind in InterfaceKind.interface_type_image():
+                civ_pks.add(civ.pk)
+                upload_pks[civ.pk] = create_upload(value)
+            elif ci.kind in InterfaceKind.interface_type_file():
+                name = get_valid_filename(value[0].name)
+                with value[0].open() as f:
+                    civ.file = File(f, name=name)
+                    civ.save()
+                civ_pks.add(civ.pk)
+            else:
+                civ.value = value
+                civ.save()
+                civ_pks.add(civ.pk)
+
+        tasks = add_values_to_archive_item.signature(
+            kwargs={
+                "archive_item_pk": self.archive_item.pk,
+                "civ_pks": list(civ_pks),
+            },
+            immutable=True,
+        )
+        if len(upload_pks) > 0:
+            image_tasks = group(
+                add_images_to_component_interface_value.signature(
+                    kwargs={
+                        "component_interface_value_pk": civ_pk,
+                        "upload_pk": upload_pk,
+                    },
+                    immutable=True,
+                )
+                for civ_pk, upload_pk in upload_pks.items()
+            )
+            tasks = chord(image_tasks, tasks)
+
+        on_commit(tasks.apply_async)
+
+        return HttpResponseRedirect(
+            reverse(
+                "archives:items-list", kwargs={"slug": self.kwargs["slug"]},
+            )
+        )
+
+
+class ArchiveItemsList(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, PaginatedTableListView,
+):
+    model = ArchiveItem
+    permission_required = (
+        f"{Archive._meta.app_label}.use_{Archive._meta.model_name}"
+    )
+    raise_exception = True
+    template_name = "archives/archive_items_list.html"
+    row_template = "archives/archive_items_row.html"
+    search_fields = [
+        "pk",
+        "name",
+    ]
+    columns = [
+        Column(title="Values", sort_field="pk"),
+        Column(title="Edit", sort_field="pk"),
+    ]
+
+    @cached_property
+    def archive(self):
+        return get_object_or_404(Archive, slug=self.kwargs["slug"])
+
+    def get_permission_object(self):
+        return self.archive
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({"archive": self.archive})
+        return context
+
+    def get_queryset(self):
+        qs = super().get_queryset()
+        return qs.filter(archive=self.archive)
```

Comment:
I think that we will need some prefetching for the values titles, did you check the debug toolbar?

Reply:
Did not, but will do!